Add dim_inactive config parameter

vihu opened the pull request:
Summary
----

This adds an optional parameter `dim_inactive` (boolean). Default value is set to `false`.

Fix #8

![2023-10-17 18-58-14](https://github.com/ribru17/bamboo.nvim/assets/3199183/0de6d6c5-23c7-48f1-9483-3fa10b34b4c6)

vihu commented on lua/bamboo/highlights.lua:
```diff
@@ -64,6 +64,10 @@ local colors = {
 
 hl.common = {
   Normal = { fg = c.fg, bg = cfg.transparent and c.none or c.bg0 },
+  NormalNC = not cfg.transparent and cfg.dim_inactive and {
```
Hmm, this would imply that you can't do both transparent=true, dim_inactive=true and honor both.

vihu commented:
Made another change [2a2a837](https://github.com/ribru17/bamboo.nvim/pull/9/commits/2a2a837139107578bee23a74945316b935afbb13) which I think looks nicer and honors both transparency and dim_inactive modes marginally better.

ribru17 commented:
Thanks so much for the PR! Sorry for the late response to your issue. I changed your code just a bit:

- Now `dim_inactive` will respect `transparent = true`
- I also made it so that both light and dark theme have slightly darker inactive windows, rather than light theme having slightly darker and dark theme having slightly lighter
- I made the `fg` for the inactive window `light_grey` instead of `grey` for slightly more contrast / readability

Let me know what you think of these changes. Thanks for taking a look!

ribru17 commented:
Oops sorry, not sure why that previous comment sent so late, I thought I sent it much earlier. Anyway thanks for the changes again I'll take a look at them now!

ribru17 commented:
Ok after taking a look again I think the previous change makes more sense; with the latest change, if someone has both `transparent = true` and `dim_inactive = true`, then only the current split will be transparent and the others will be fully opaque. I think this might not be fully desirable. Do you know if there is a way to have a slightly darker transparent color (or maybe semitransparent)? The change I implemented just copied `tokyonight.nvim`, as you saw: fully ignoring the `dim_inactive` setting when transparency is set.

vihu commented:
> Ok after taking a look again I think the previous change makes more sense; with the latest change, if someone has both `transparent = true` and `dim_inactive = true`, then only the current split will be transparent and the others will be fully opaque. I think this might not be fully desirable. Do you know if there is a way to have a slightly darker transparent color (or maybe semitransparent)? The change I implemented just copied `tokyonight.nvim`, as you saw: fully ignoring the `dim_inactive` setting when transparency is set.

I think it really is individual preference, visually (at least to my eyes) the current version looks more distinct. I don’t quite know how to make it work exactly as you mention while honoring both transparency and dim_inactive. In my local config I do set both to true and it looks reasonably well.

FWIW I borrowed the changes from [here](https://github.com/ellisonleao/gruvbox.nvim/pull/149) (I implemented the same thing on gruvbox heh).

I _think_ kanagawa implements it slightly differently, might be worth taking a look there as well. Unfortunately I’m no color theory expert, I basically just wanted to put this PR up for your reference, eventually its your call :)

ribru17 commented:
Ok sounds good, let's keep it like that then :) If people want it the other way in the future I guess we can always add some hacky option to make all buffers transparent even when dim is set (but at that point why would dim be set I guess? :D). Also I tweaked one or two more things if that's alright, to reduce redundancy and make sure that floating windows look nice in transparent mode. Thanks for the help!
